pkg/checkoutinfo/checkoutinfo.go

```diff
@@ -0,0 +1,136 @@
+package checkoutinfo
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/pkg/buildscript"
+)
+
+type ErrInvalidCommitID struct {
+	CommitID string
+}
+
+func (e ErrInvalidCommitID) Error() string {
+	return "invalid commit ID"
+}
+
+type projectfiler interface {
+	Owner() string
+	Name() string
+	BranchName() string
+	LegacyCommitID() string
+	SetNamespace(string, string) error
+	SetBranch(string) error
+	SetLegacyCommit(string) error
+	Dir() string
+	URL() string
+}
+
+type configurer interface {
+	GetBool(string) bool
+}
+
+type CheckoutInfo struct {
+	project projectfiler
+	config  configurer
+}
+
+var ErrBuildscriptNotExist = errors.New("Build script does not exist")
+
+func New(project projectfiler, config configurer) *CheckoutInfo {
+	return &CheckoutInfo{project, config}
+}
+
+// Owner returns the project owner from activestate.yaml.
+// Note: cannot read this from buildscript because it may not exist yet.
+func (c *CheckoutInfo) Owner() string {
+	return c.project.Owner()
```

Comment:
Please update all the project methods as legacy, same as we did with the commitID. That extends to the `Set..()` ones also.

Reply:
You said in the previous PR that not everything should go into the checkoutinfo package. If we want to work with projects, we use the project package. If we want to work with buildscripts, we use the buildscript package. By making everything legacy, we're going to be forcing ourselves into buildscripts so that we end up with `prime.buildscript.Owner()`, etc. instead of `prime.project.Owner()`, etc.

That said, I have serious reservations about this. Not only is it going to balloon the PR by 4x, but I can see us walking back on this and undoing it is going to be really messy. We already walked back the CommitID() -> LegacyCommitID() transition. (That is, we went from `project.CommitID()` to `localcommit.Get()` and back to `project.LegacyCommitID()` via `localcommit.Get()`.) It's a mess.

As I've been working on this PR, I've been thinking that if build script parse fails, we have no clue which project we're working on. `state reset LOCAL` cannot fix it because we need to parse it to get the project info. Since build scripts are far more brittle than activestate.yaml, this will probably happen pretty frequently as users make edits, or as build scripts change format over time.

As an example, start with a v46 build script and try to do something with this PR. It'll fail with a parse error. There's a lot we'll have to do to make it more robust, especially if we end up priming the primer with a buildscript instead of a projectfile.

Finally, my editor doesn't have method refactoring, so I'd either have to do this by hand or figure out how to use a new editor that has such functionality. If you really want this, can we make it a new ticket? :)

Reply:
Let's chat on Slack when you're around to figure out the approach. I totally feel you on this starting to feel messy. It's worth having another look at our use-case and seeing if this is in fact the right solution, before we commit to it.

The refactor itself isn't much of an issue here though, with tooling that's really easy to do. I'm more worrying about if this is actually going to save us pain during the transition from as.yaml > buildscripts, rather than cause pain.